### PR TITLE
feat: add logic to track trades as sales in eth and polygon

### PR DESCRIPTION
**Pull request by juanmahidalgo**
This PR adds logic to track the `Traded` event from the new offchain marketplace contract. 
It will track events from both Ethereum and Polygon and track the trades executed as sales and persist all the data relevant to the event.

### CODE REVIEW

File: src/polygon/handlers/marketplace.ts [hunk 1]
```diff
@@ -216,3 +220,109 @@ export function handleOrderCancelled(
...
+    ctx,
+    block.header,
+    addresses.MarketplaceV3
+  );
+  const feesCollector = await marketplaceV3Contract.feeCollector();
+  const feeRate = await marketplaceV3Contract.feeRate();
+  const royaltiesRate = await marketplaceV3Contract.royaltiesRate();
+
+  // NFT
+  if (assetType === 3) {
```
Comment (cyaiox) [inline src/polygon/handlers/marketplace.ts:248]: do you think it makes sense to store the assetType = 3 || 4 into an enum?

Comment (nachomazzara) [reply src/polygon/handlers/marketplace.ts:248]: +1

File: src/common/utils/marketplaceV3.ts [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  const contractAddressReceived = event._trade.received[0].contractAddress;
+
+  // if received is MANA, then it's an order. We could also have other ERC20 sent, but for the moment we are only checking for MANA
+  if (
+    isERC20Asset &&
+    [addresses.MANA, addresses.TRANSAK_TOKEN].includes(contractAddressReceived) // support Transak token to track sales in dev
+  ) {
+    return TradeType.Order;
+  }
+  return TradeType.Bid;
```
Comment (nachomazzara) [inline src/common/utils/marketplaceV3.ts:34]: If it is not MANA, we should skip it, right? fallback to Bid is weird

File: src/polygon/handlers/marketplace.ts [hunk 3]
```diff
@@ -337,6 +337,6 @@ export async function handleTraded(
       event
     );
   } else {
-    console.log("ERROR: NFT not found for sale in traded event");
+    console.log("ERROR: Asset type not supported in trade: ", assetType);
```
Comment (nachomazzara) [inline src/polygon/handlers/marketplace.ts:340]: Log the tx hash or trade payload at least?

**Review by nachomazzara — APPROVED** [review]

**Pull request merged**